File: app/src/main/java/com/pasiflonet/mobile/SettingsActivity.java
package com.pasiflonet.mobile;

import android.app.Activity;
import android.os.Bundle;
import android.text.InputType;
import android.widget.*;

import com.pasiflonet.mobile.ui.UltraUi;
import com.pasiflonet.mobile.util.Prefs;

public class SettingsActivity extends Activity {
    EditText channel, watermark, botToken, botChat, apiId, apiHash;
    CheckBox botFallback;

    @Override public void onCreate(Bundle b) {
        super.onCreate(b);
        UltraUi.setup(this);
        LinearLayout root = new LinearLayout(this);
        setContentView(UltraUi.page(this, root));
        root.addView(UltraUi.title(this, "הגדרות Azretr Ultra"));
        root.addView(UltraUi.sub(this, "הפרטים נשמרים מקומית במכשיר. לא לשים כאן טוקנים אם המכשיר לא מאובטח."));

        apiId = UltraUi.input(this, "api_id של Telegram"); apiId.setInputType(InputType.TYPE_CLASS_NUMBER); apiId.setText(Prefs.get(this,"api_id", ""));
        apiHash = UltraUi.input(this, "api_hash של Telegram"); apiHash.setText(Prefs.get(this,"api_hash", ""));
        channel = UltraUi.input(this, "ערוץ יעד TDLib — למשל @myChannel"); channel.setText(Prefs.get(this,"channel_username", ""));
        watermark = UltraUi.input(this, "סימן מים ברירת מחדל"); watermark.setText(Prefs.get(this,"watermark", "Azretr Ultra"));
        botToken = UltraUi.input(this, "Bot Token אופציונלי לגיבוי"); botToken.setInputType(InputType.TYPE_CLASS_TEXT | InputType.TYPE_TEXT_VARIATION_PASSWORD); botToken.setText(Prefs.get(this,"bot_token", ""));
        botChat = UltraUi.input(this, "Chat ID / @channel לגיבוי Bot API"); botChat.setText(Prefs.get(this,"bot_chat", ""));
        botFallback = new CheckBox(this); botFallback.setText("אפשר מצב גיבוי Bot API לשליחה"); botFallback.setTextColor(UltraUi.TEXT); botFallback.setTextSize(16); botFallback.setChecked(Prefs.getBool(this,"bot_fallback", false));

        LinearLayout c = UltraUi.card(this);
        c.addView(UltraUi.text(this, "Telegram / TDLib", 20, UltraUi.TEXT, true));
        c.addView(apiId); c.addView(apiHash); c.addView(channel); root.addView(c);
        LinearLayout m = UltraUi.card(this);
        m.addView(UltraUi.text(this, "מדיה ושליחה", 20, UltraUi.TEXT, true));
        m.addView(watermark); m.addView(botFallback); m.addView(botToken); m.addView(botChat); root.addView(m);

        Button save = UltraUi.button(this, "שמור הגדרות");
        save.setOnClickListener(v -> {
            Prefs.set(this,"api_id", apiId.getText().toString().trim());
            Prefs.set(this,"api_hash", apiHash.getText().toString().trim());
            Prefs.set(this,"channel_username", channel.getText().toString().trim());
            Prefs.set(this,"watermark", watermark.getText().toString());
            Prefs.set(this,"bot_token", botToken.getText().toString().trim());
            Prefs.set(this,"bot_chat", botChat.getText().toString().trim());
            Prefs.setBool(this,"bot_fallback", botFallback.isChecked());
            Toast.makeText(this, "נשמר", Toast.LENGTH_SHORT).show();
        });
        root.addView(save);
    }
}

File: app/src/main/java/com/pasiflonet/mobile/SettingsActivity.kt
package com.pasiflonet.mobile

import android.content.SharedPreferences
import android.net.Uri
import android.os.Bundle
import android.widget.Toast
import androidx.activity.result.contract.ActivityResultContracts
import com.pasiflonet.mobile.databinding.ActivitySettingsBinding
import java.io.File
import java.io.FileOutputStream

class SettingsActivity : BaseActivity() {
    private lateinit var b: ActivitySettingsBinding
    private lateinit var prefs: SharedPreferences

    private val pickLogo = registerForActivityResult(ActivityResultContracts.GetContent()) { uri: Uri? ->
        if (uri != null) {
            try {
                val inputStream = contentResolver.openInputStream(uri)
                val localFile = File(filesDir, "channel_logo.png")
                val outputStream = FileOutputStream(localFile)
                inputStream?.copyTo(outputStream)
                inputStream?.close()
                outputStream.close()
                val localUri = Uri.fromFile(localFile).toString()
                prefs.edit().putString("logo_uri", localUri).apply()
                b.ivCurrentLogo.setImageURI(Uri.parse(localUri))
                Toast.makeText(this, "הלוגו נשמר", Toast.LENGTH_SHORT).show()
            } catch (e: Exception) {
                Toast.makeText(this, "שגיאה: ${e.message}", Toast.LENGTH_LONG).show()
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        try {
            b = ActivitySettingsBinding.inflate(layoutInflater)
            setContentView(b.root)
            prefs = getSharedPreferences("app_prefs", MODE_PRIVATE)

            b.etTargetUsername.setText(prefs.getString("target_username", "") ?: "")
            b.etSignature.setText(prefs.getString("text_signature", "") ?: "")
            b.etKeywords.setText(prefs.getString("alert_keywords", "") ?: "")

            val currentLogo = prefs.getString("logo_uri", "") ?: ""
            if (currentLogo.isNotEmpty()) {
                try { b.ivCurrentLogo.setImageURI(Uri.parse(currentLogo)) } catch (_: Exception) {}
            }

            b.btnSaveSettings.setOnClickListener {
                val target = b.etTargetUsername.text?.toString()?.trim().orEmpty()
                val signature = b.etSignature.text?.toString()?.trim().orEmpty()
                val keywords = b.etKeywords.text?.toString()?.trim().orEmpty()

                prefs.edit()
                    .putString("target_username", target)
                    .putString("text_signature", signature)
                    .putString("alert_keywords", keywords)
                    .apply()

                Toast.makeText(this, "ההגדרות נשמרו", Toast.LENGTH_SHORT).show()
                finish()
            }

            b.btnSelectLogo.setOnClickListener { pickLogo.launch("image/*") }

            b.btnClearCache.setOnClickListener {
                try {
                    cacheDir.deleteRecursively()
                    Toast.makeText(this, "הקאש נוקה", Toast.LENGTH_SHORT).show()
                } catch (_: Exception) {}
            }
        } catch (e: Exception) {
            Toast.makeText(this, "שגיאה בפתיחת הגדרות: ${e.message}", Toast.LENGTH_LONG).show()
            finish()
        }
    }
}
